File: DAY 5/LinkedList.java
public class LinkedList {
    // Node class representing each element in the list
    static class Node {
        int data;
        Node next;

        // Constructor
        Node(int data) {
            this.data = data;
            this.next = null;
        }
    }

    public static void main(String[] args) {
        // Create individual nodes
        Node head = new Node(10);
        Node middle = new Node(20);
        Node last = new Node(30);

        // Link the nodes together
        head.next = middle;
        middle.next = last;

        // Print the linked list
        System.out.print("Linked List: ");
        Node temp = head;
        while (temp != null) {
            System.out.print(temp.data + " -> ");
            temp = temp.next;
        }
        System.out.println("NULL");
    }
}

File: DAY 5/LinkedList.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    // Define a struct for linked list node
    struct node {
        int data;
        struct node *next;
    };

    // Create three nodes
    struct node *head, *middle, *last;

    // Allocate memory for each node
    head = (struct node *)malloc(sizeof(struct node));
    middle = (struct node *)malloc(sizeof(struct node));
    last = (struct node *)malloc(sizeof(struct node));

    // Assign data to each node
    head->data = 10;
    middle->data = 20;
    last->data = 30;

    // Link the nodes
    head->next = middle;
    middle->next = last;
    last->next = NULL;  

    // Temporary pointer to traverse the list
    struct node *temp = head;

    // Traverse and print the linked list
    printf("Linked List: ");
    while (temp != NULL) {
        printf("%d -> ", temp->data);
        temp = temp->next;
    }
    printf("NULL\n");

    // Free the allocated memory
    free(head);
    free(middle);
    free(last);

    return 0;
}
